src/nvim/eval.c

```diff
@@ -5177,6 +5265,23 @@ void list_append_string(list_T *l, char_u *str, int len)
   }
 }
 
+/// Append given string to the list
+///
+/// Unlike list_append_string this function does not copy the string.
+///
+/// @param[out]  l    List to append to.
+/// @param[in]   str  String to append.
+void list_append_allocated_string(list_T *l, char *const str)
+  FUNC_ATTR_NONNULL_ARG(1)
```

Comment:
`list_append_string` can call `list_append_allocated_string()` instead of repeating the logic.